utils/log/logger.go

```diff
@@ -0,0 +1,184 @@
+package log
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/golang/glog"
+)
+
+// DebugEnabled specifies if this package print debug information
+var DebugEnabled = false
+
+// Logger is a struct which will help to call CITF specific logging functions
+type Logger struct{}
```

Comment:
@a4abhishek I don't see point in reinventing the wheel! 
Why can't you use `glog` alone for logging purpose and adjust the logic for `DebugEnabled` with it?

Reply:
@satyamz we were doing same, but 
- with earlier approach if I want to log something only when `DebugEnabled` is `true` we had to write at least 3 lines **now just one line is enough**.
- with earlier approach if I want to log error when `error` is not `nil` and `DebugEnabled` is `true`, we had to write at lease 5 lines **now just one line is enough**.
- `Bettercodehug` is pushing us to modularize the code to its limit, and I feel it will be a little help. (at least it will help me in NDM)
- If you ignore `logger.go` In this package alone **I saved 60 lines of code**

It is not really reinventing the wheel, it is more like customizing it with vulcanized rubber tires. :smile:

Reply:
@satyamz @a4abhishek good effort. Its good to improvise. 

However, most of the times it is the _**first conclusion bias**_ that leads us to implementing the first thoughts that we encounter. Lets try to focus on contrarian opinions / further queries:

- Is the code usage idiomatic?
- Does it mandate the developer into thinking in some specific way which in turn may not be idiomatic, lead to learning curve, less community contribution & so on?
- Is the code based on the foundations of best programming practices?

I will try to conclude with an example of error handling. You may want to review it with above questions or more questions you can come up with & then decide. This is not a code review, but my general thought.

```go
type Processor interface {
	Process(buf []byte) (message string, err error)
}

type ErrorHandler interface {
	Handle(err error)
}

func RegularError(buf []byte, processor Processor,
	handler ErrorHandler) (message string, err error) {
	message, err = processor.Process(buf)
	if err != nil {
		handler.Handle(err)
		return "", err
	}
	return
}
```

Reply:
Thanks @AmitKumarDas for input.